### PR TITLE
Lowercase TargetOS

### PR DESCRIPTION
**Pull request by ViktorHofer**
Follow-up on https://github.com/dotnet/runtime/issues/32451#issuecomment-634937878
Follow-up on https://github.com/dotnet/runtime/pull/80074

In https://github.com/dotnet/runtime/pull/43651, the "Windows" OS name was lowercased to achieve target frameworks with RIDs in them. At that time, the original plan was to lowercase all TargetOS values but that had to be cut because of resources. This PR finishes that and updates build systems, YML, markdowns and managed components.

**To not break existing developer workflows, the passed-in `-os` value is automatically lower-cased in the script entry points and validated in the msbuild entry point.**

**This change was performed via**
1. Manual "Find & Replace"**: `<OS>` -> `<os>` for all the possible OS values that we support (i.e. Browser -> browser) in msbuild, cmake, docs, etc.
2. Trial & Error to fix build failures
3. Looking through all the commits that went in between the Find & Replace was performed and now (Jan 10th) and performing the same operation again

### CODE REVIEW

File: eng/native/build-commons.sh [hunk 1]
```diff
@@ -236,12 +236,12 @@ __BuildOS=$os
 __OutputRid=''
 
 # Get the number of processors available to the scheduler
-platform="$(uname)"
-if [[ "$platform" == "FreeBSD" ]]; then
+platform="$(uname -s | tr '[A-Z]' '[a-z]')"
```
Comment (akoeplinger) [inline eng/native/build-commons.sh:239]: for consistency with other usages:

```suggestion
platform="$(uname -s | tr '[:upper:]' '[:lower:]')"
```

File: eng/native/gen-buildsys.sh [hunk 2]
```diff
@@ -61,9 +61,9 @@ done
 
 cmake_extra_defines=
 if [[ "$CROSSCOMPILE" == "1" ]]; then
-    platform="$(uname)"
+    platform="$(uname -s | tr '[A-Z]' '[a-z]')"
```
Comment (akoeplinger) [inline eng/native/gen-buildsys.sh:64]:
```suggestion
    platform="$(uname -s | tr '[:upper:]' '[:lower:]')"
```

File: eng/native/init-os-and-arch.sh [hunk 3]
```diff
@@ -1,17 +1,17 @@
 #!/usr/bin/env bash
 
 # Use uname to determine what the OS is.
-OSName=$(uname -s)
+OSName=$(uname -s | tr '[A-Z]' '[a-z]')
```
Comment (akoeplinger) [inline eng/native/init-os-and-arch.sh:4]:
```suggestion
OSName=$(uname -s | tr '[:upper:]' '[:lower:]')
```

File: src/mono/CMakeLists.txt [hunk 4]
```diff
@@ -182,7 +182,7 @@ if(CMAKE_SYSTEM_NAME STREQUAL "Darwin")
   set(HOST_OSX 1)
   set(PTHREAD_POINTER_ID 1)
   set(USE_MACH_SEMA 1)
-  if(CMAKE_SYSTEM_VARIANT STREQUAL "MacCatalyst")
+  if(CMAKE_SYSTEM_VARIANT STREQUAL "maccatalyst")
     set(HOST_MACCAT 1)
   endif()
 elseif(CMAKE_SYSTEM_NAME STREQUAL "iOS" OR CMAKE_SYSTEM_NAME STREQUAL "tvOS")
```
Comment (akoeplinger) [inline src/mono/CMakeLists.txt:188]: can you please fix this as well, by introducing `CLR_CMAKE_HOST_OS` a few lines above? longer term I'd like to unify the separate Mono cmake files with configureplatform.cmake

**Review by akoeplinger — APPROVED** [review]

**Review by akoeplinger — APPROVED** [review]

**Comment by EgorBo** [comment]
Just updated my local branch:

image

I assume it needs `git clean -ffddxx` ?

**Comment by ViktorHofer** [comment]
Sorry for the inconvenience and thanks for reporting. Just submitted a fix: https://github.com/dotnet/runtime/pull/80544

**Pull request merged**